src/core/Akka/Actor/FSM.cs

```diff
@@ -795,24 +886,14 @@ public bool IsTimerActive(string name)
         /// <param name="timeout">TBD</param>
         public void SetStateTimeout(TState state, TimeSpan? timeout)
         {
-            if(!_stateTimeouts.ContainsKey(state))
-                _stateTimeouts.Add(state, timeout);
-            else
-                _stateTimeouts[state] = timeout;
+            _stateTimeouts.AddOrSet(state, timeout);
```

Comment:
Change this to 

`stateTimeouts[state] = timeout`

AddOrSet is something stupid I added to the codebase that we should deprecate and remove at some point. It's an unnecessary abstraction.